src/entities/BlockNode/index.ts

```diff
@@ -64,10 +82,25 @@ export class BlockNode {
    * Returns serialized object representing the BlockNode
    */
   public get serialized(): BlockNodeSerialized {
+    const map = (data: BlockNodeDataValue): BlockNodeDataSerializedValue => {
+      if (Array.isArray(data)) {
+        return data.map(map) as BlockNodeDataSerialized[];
+      }
+
+      if (data instanceof ValueNode || data instanceof TextNode) {
+        return data.serialized;
+      }
+
+      return Object.fromEntries(
+        Object.entries(data)
+          .map(([key, value]) => ([key, map(value)]))
+      );
```

Comment:
looks hard to read, maybe it would be better to move it to utility with a nice name?